clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -480,14 +480,17 @@ ProgramInfo::insertIntoExternalFunctionMap(ExternalFunctionMapType &Map,
     RetVal = true;
   } else {
     // MultipleRewrites = true;
+    auto oldS = Map[FuncName];
+    auto *newC = getOnly(ToIns);
+    auto *oldC = getOnly(oldS);
     if (isDef) {
-      auto oldS = Map[FuncName];
-      auto *newC = getOnly(ToIns);
-      auto *oldC = getOnly(oldS);
-      // Retain CVars and argConstraints from old ConstraintVariable
       newC->brainTransplant(oldC);
       Map[FuncName] = ToIns;
       RetVal = true;
+    } else if (!oldC->hasBody()) {
+      // if the current FV constraint is not a definition?
+      // then merge.
+      oldC->mergeDeclaration(newC);
```

Comment:
Per my comments above, you are merging the new into the old. Therefore, you should not be blowing away the old `argumentConstraints`; you should just leave them and ignore the new ones (which should be empty). And you should merge the `itypeStr` (I think---don't know how these work).